### DISCUSSION TITLE
Error After Update Weewx to 4.7.0

### DISCUSSION THREAD

**Discussion by Overcon** [poblabs/weewx-belchertown#774] [Q&A] [answered]
Last Sunday i updated Weewx to version 4.7 and I thought all was well, but I just noticed that while MQTT is updating the temperature when you visit the page, the graphs and charts are not updating. In the syslog, I see the following error, can anyone help me figure out what is erroring out to resolve? I can provide additional logs, just let me know which ones.

Thanks! I would like to figur eout what is erroring and get my stats back up and running.

I also updated the belchertown skin to the latest master version in case it was something on the skin, nothing changed.

~:weewx-pi@userx$sudo tail -f /var/log/syslog
[sudo] password for userx:
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.restx: AWEKAS: wait interval (120 < 600) has not passed for record 2022-03-29 19:02:00 MDT (1648602120)
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.restx: WeatherCloud: wait interval (120 < 600) has not passed for record 2022-03-29 19:02:00 MDT (1648602120)
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.restx: OWM: wait interval (120 < 600) has not passed for record 2022-03-29 19:02:00 MDT (1648602120)
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.reportengine: Running reports for latest time in the database.
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.reportengine: Running report 'SeasonsReport'
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.restx: Twitter: wait interval (120 < 3600) has not passed for record 2022-03-29 19:02:00 MDT (1648602120)
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.reportengine: Found configuration file /home/weewx/skins/Seasons/skin.conf for report 'SeasonsReport'
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.cheetahgenerator: Using search list ['weewx.cheetahgenerator.Almanac', 'weewx.cheetahgenerator.Current', 'weewx.cheetahgenerator.DisplayOptions', 'weewx.cheetahgenerator.Extras', 'weewx.cheetahgenerator.Gettext', 'weewx.cheetahgenerator.JSONHelpers', 'weewx.cheetahgenerator.PlotInfo', 'weewx.cheetahgenerator.SkinInfo', 'weewx.cheetahgenerator.Station', 'weewx.cheetahgenerator.Stats', 'weewx.cheetahgenerator.UnitInfo']
Mar 29 19:02:30 weewx-pi weewx[876] INFO weewx.restx: MQTT: Published record 2022-03-29 19:02:00 MDT (1648602120)
Mar 29 19:02:30 weewx-pi weewx[876] DEBUG weewx.manager: Daily summary version is 4.0
Mar 29 19:02:33 weewx-pi weewx[876] INFO weewx.cheetahgenerator: Generated 8 files for report SeasonsReport in 3.30 seconds
Mar 29 19:02:33 weewx-pi weewx[876] DEBUG weewx.manager: Daily summary version is 4.0
Mar 29 19:02:34 weewx-pi weewx[876] INFO weewx.imagegenerator: Generated 15 images for report SeasonsReport in 0.89 seconds
Mar 29 19:02:34 weewx-pi weewx[876] INFO weewx.reportengine: Copied 0 files to /var/www/lovingtonweather.com/weewx
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.reportengine: Report 'SmartphoneReport' not enabled. Skipping.
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.reportengine: Report 'MobileReport' not enabled. Skipping.
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.reportengine: Report 'StandardReport' not enabled. Skipping.
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.reportengine: Running report 'Highcharts'
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.reportengine: Found configuration file /home/weewx/skins/Highcharts/skin.conf for report 'Highcharts'
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.cheetahgenerator: Using search list ['user.highchartssearchlist.HighchartsMinRanges', 'user.highchartssearchlist.HighchartsWeek', 'user.highchartssearchlist.HighchartsYear', 'user.highchartssearchlist.HighchartsWindRose', 'weewx.cheetahgenerator.Almanac', 'weewx.cheetahgenerator.Current', 'weewx.cheetahgenerator.DisplayOptions', 'weewx.cheetahgenerator.Extras', 'weewx.cheetahgenerator.Gettext', 'weewx.cheetahgenerator.JSONHelpers', 'weewx.cheetahgenerator.PlotInfo', 'weewx.cheetahgenerator.SkinInfo', 'weewx.cheetahgenerator.Station', 'weewx.cheetahgenerator.Stats', 'weewx.cheetahgenerator.UnitInfo']
Mar 29 19:02:34 weewx-pi weewx[876] DEBUG weewx.manager: Daily summary version is 4.0
Mar 29 19:02:36 weewx-pi weewx[876] DEBUG weewx.cheetahgenerator: Skip 'year.json': last_mod=1648601797.563521 age=359.32245445251465 stale=3600
Mar 29 19:02:36 weewx-pi weewx[876] INFO weewx.cheetahgenerator: Generated 1 files for report Highcharts in 2.46 seconds
Mar 29 19:02:36 weewx-pi weewx[876] DEBUG weewx.reportengine: Running report 'Belchertown'
Mar 29 19:02:36 weewx-pi weewx[876] DEBUG weewx.reportengine: Found configuration file /home/weewx/skins/Belchertown/skin.conf for report 'Belchertown'
Mar 29 19:02:36 weewx-pi weewx[876] DEBUG weewx.cheetahgenerator: Using search list ['user.belchertown.getData', 'weewx.cheetahgenerator.Almanac', 'weewx.cheetahgenerator.Current', 'weewx.cheetahgenerator.DisplayOptions', 'weewx.cheetahgenerator.Extras', 'weewx.cheetahgenerator.Gettext', 'weewx.cheetahgenerator.JSONHelpers', 'weewx.cheetahgenerator.PlotInfo', 'weewx.cheetahgenerator.SkinInfo', 'weewx.cheetahgenerator.Station', 'weewx.cheetahgenerator.Stats', 'weewx.cheetahgenerator.UnitInfo']
Mar 29 19:02:36 weewx-pi weewx[876] DEBUG weewx.manager: Daily summary version is 4.0
Mar 29 19:02:40 weewx-pi weewx[876] INFO weewx.cheetahgenerator: Generated 11 files for report Belchertown in 3.59 seconds
Mar 29 19:02:40 weewx-pi weewx[876] INFO weewx.reportengine: Copied 2 files to /var/www/lovingtonweather.com/
Mar 29 19:02:40 weewx-pi weewx[876] DEBUG weewx.manager: Daily summary version is 4.0
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine: Caught unrecoverable exception in generator 'user.belchertown.HighchartsJsonGenerator'
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****  invalid literal for int() with base 10: '%'
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****  Traceback (most recent call last):
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****    File "/home/weewx/bin/weewx/reportengine.py", line 197, in run
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****      obj.start()
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****    File "/home/weewx/bin/weewx/reportengine.py", line 378, in start
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****      self.run()
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****    File "/home/weewx/bin/user/belchertown.py", line 2714, in run
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****      series_data = self.get_observation_data(
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****    File "/home/weewx/bin/user/belchertown.py", line 3726, in get_observation_data
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****      usage_round = int(
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****  ValueError: invalid literal for int() with base 10: '%'
Mar 29 19:02:40 weewx-pi weewx[876] ERROR weewx.reportengine:         ****  Generator terminated
Mar 29 19:02:40 weewx-pi weewx[876] DEBUG weewx.reportengine: Report 'FTP' not enabled. Skipping.
Mar 29 19:02:40 weewx-pi weewx[876] DEBUG weewx.reportengine: Report 'RSYNC' not enabled. Skipping.
Mar 29 19:02:49 weewx-pi weewx[876] DEBUG user.gw1000: Next update in 20 seconds
Mar 29 19:02:49 weewx-pi weewx[876] DEBUG weewx.restx: Wunderground-RF: wait interval (21 < 80) has not passed for record 2022-03-29 19:02:49 MDT (1648602169)
Mar 29 19:02:49 weewx-pi weewx[876] INFO weewx.restx: MQTT: Published record 2022-03-29 19:02:49 MDT (1648602169)

**Comment by Overcon** [comment]
Here are the Weewx conf and skin conf files hopefully I redacted anything important :) Well, weather data important anyway.

Really appreciate any help.

[weewx.txt](https://github.com/poblabs/weewx-belchertown/files/8379542/weewx.txt)
[skin.txt](https://github.com/poblabs/weewx-belchertown/files/8379544/skin.txt)

**Reply by James-76** [reply]
This looks similar to what was discussed in #761 and it is due to values not having units correctly defined so graphs.conf breaks

I'm sure there is a way to improve the exception code here, but as I don't have extras like lighting count it was difficult for me to dive any more into the code.

James

**Comment by James-76** [comment, marked as answer]
OK

I ran some debug in belchertown.py and asked it to print out 

```
        print(observation)
        print(obs_vt)
        print(self.skin_dict["Units"]["StringFormats"])
```

It crashes out because of two reasons
- count is not defined as a Floating point decimal format but as a signed integer decimal
- The default aggregate_type = None is applied to the lightning_strike_count

```
lightning_strike_count
([None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None], 'count', 'group_count')
{'amp': '%.1f', 'bit': '%.0f', 'boolean': '%d', 'byte': '%.0f', 'centibar': '%.0f', 'cm': '%.2f', 'cm_per_hour': '%.2f', 'count': '%d', 'cubic_foot': '%.1f', 'day': '%.1f', 'dB': '%.0f', 'degree_C': '%.1f', 'degree_C_day': '%.1f', 'degree_compass': '%.0f', 'degree_E': '%.1f', 'degree_F': '%.1f', 'degree_F_day': '%.1f', 'degree_K': '%.1f', 'foot': '%.0f', 'gallon': '%.1f', 'hour': '%.1f', 'hPa': '%.1f', 'hPa_per_hour': '%.3f', 'inch': '%.2f', 'inch_per_hour': '%.2f', 'inHg': '%.3f', 'inHg_per_hour': '%.5f', 'kilowatt': '%.1f', 'kilowatt_hour': '%.1f', 'km': '%.1f', 'km_per_hour': '%.0f', 'km_per_hour2': '%.1f', 'knot': '%.0f', 'knot2': '%.1f', 'kPa': '%.2f', 'kPa_per_hour': '%.4f', 'liter': '%.1f', 'litre': '%.1f', 'lux': '%.0f', 'mbar': '%.1f', 'mbar_per_hour': '%.4f', 'mega_joule': '%.0f', 'meter': '%.0f', 'meter_per_second': '%.1f', 'meter_per_second2': '%.1f', 'microgram_per_meter_cubed': '%.0f', 'mile': '%.1f', 'mile_per_hour': '%.0f', 'mile_per_hour2': '%.1f', 'minute': '%.1f', 'mm': '%.1f', 'mm_per_hour': '%.1f', 'mmHg': '%.1f', 'mmHg_per_hour': '%.4f', 'percent': '%.0f', 'ppm': '%.0f', 'second': '%.0f', 'uv_index': '%.1f', 'volt': '%.1f', 'watt': '%.1f', 'watt_hour': '%.1f', 'watt_per_meter_squared': '%.0f', 'watt_second': '%.0f', 'NONE': '   N/A'}
```

I solved this by adding the following to [[StringFormats]] in your Belchertown skin.conf

```
    [[StringFormats]]
        # This section sets the string formatting for each type of unit.
...
        watt_per_meter_squared = %.0f
        count              = %.0f
        NONE               = "N/A"
```


```
lightning_strike_count
([None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None], 'count', 'group_count')
{'amp': '%.1f', 'bit': '%.0f', 'boolean': '%d', 'byte': '%.0f', 'centibar': '%.0f', 'cm': '%.2f', 'cm_per_hour': '%.2f', 'count': '%.0f', 'cubic_foot': '%.1f', 'day': '%.1f', 'dB': '%.0f', 'degree_C': '%.1f', 'degree_C_day': '%.1f', 'degree_compass': '%.0f', 'degree_E': '%.1f', 'degree_F': '%.1f', 'degree_F_day': '%.1f', 'degree_K': '%.1f', 'foot': '%.0f', 'gallon': '%.1f', 'hour': '%.1f', 'hPa': '%.1f', 'hPa_per_hour': '%.3f', 'inch': '%.2f', 'inch_per_hour': '%.2f', 'inHg': '%.3f', 'inHg_per_hour': '%.5f', 'kilowatt': '%.1f', 'kilowatt_hour': '%.1f', 'km': '%.1f', 'km_per_hour': '%.0f', 'km_per_hour2': '%.1f', 'knot': '%.0f', 'knot2': '%.1f', 'kPa': '%.2f', 'kPa_per_hour': '%.4f', 'liter': '%.1f', 'litre': '%.1f', 'lux': '%.0f', 'mbar': '%.1f', 'mbar_per_hour': '%.4f', 'mega_joule': '%.0f', 'meter': '%.0f', 'meter_per_second': '%.1f', 'meter_per_second2': '%.1f', 'microgram_per_meter_cubed': '%.0f', 'mile': '%.1f', 'mile_per_hour': '%.0f', 'mile_per_hour2': '%.1f', 'minute': '%.1f', 'mm': '%.1f', 'mm_per_hour': '%.1f', 'mmHg': '%.1f', 'mmHg_per_hour': '%.4f', 'percent': '%.0f', 'ppm': '%.0f', 'second': '%.0f', 'uv_index': '%.1f', 'volt': '%.1f', 'watt': '%.1f', 'watt_hour': '%.1f', 'watt_per_meter_squared': '%.0f', 'watt_second': '%.0f', 'NONE': '   N/A'}
```

@dtalens you might be interested in this

James

**Reply by dtalens** [reply]
It's working without erros. Thanks @James-76

**Reply by James-76** [reply]
Thanks for confirming - I will submit an update to belchertown.py to see if it can handle the exception better.

James

**Reply by Overcon** [reply]
Thank you James! Much appreciated.

**Comment by James-76** [comment]
#775 has been merged in which stops the script crashing out.
